third-party/lemon/patches/allocator-patch.patch

```diff
@@ -0,0 +1,298 @@
+# HG changeset patch
```

Comment:
Is there a new lemon release we should upgrade to rather than apply one patch?

Reply:
The last lemon release was in 2014. There are commits after this point, but this patch was never merged in.

Reply:
Also important to note that this patch is only applied in the action not the DependencyInstaller script which is intentional.

Really my goal was to get the CI action up and running and not to support c++20 in the project directly.

Reply:
Does it make sense to fork that project and apply the patch there instead? It doesn't seem like they are maintaining that project and if it's a required dependency we need the ability to fix issues like this in a better way than a patch getting applied during an install step

Reply:
That sounds reasonable.  I'm sure there will be other users of c++20 - eventually even the project itself.

Reply:
Are we suggesting dumping the files in this repo, or adding a fork?

Reply:
I would make a new repo in GH.

Reply:
I'm a little bit torn on this particular issues the LEMON mercurial repo does have recent commits, but no new releases. 

I'm going to open a separate issue where we can decide what we want to do.

http://lemon.cs.elte.hu/hg/lemon

Reply:
I guess my point is this patch was proposed to them 3 years and it's still not in any release. I see the commits, but they don't appear to be that responsive. Please reference the issue when created here.

Reply:
https://github.com/The-OpenROAD-Project/OpenROAD/issues/2791

Reply:
Discussion happening in https://github.com/The-OpenROAD-Project/OpenROAD/issues/2791